packages/odyssey-react-mui/src/Autocomplete.tsx

```diff
@@ -349,19 +344,49 @@ const Autocomplete = <
     ],
   );
 
-  const renderVirtualizedRow = ({
+  const Row = ({
     data,
     index,
+    setItemSize,
     style,
-  }: ListChildComponentProps) => {
+  }: ListChildComponentProps & { setItemSize: SetItemSize }) => {
+    const rowRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+      if (rowRef.current) {
+        /**
+         * Checking for child height to workaround a bug where the clientHeight of the row isn't updated correctly
+         * @see here if you need to know more: https://github.com/bvaughn/react-window/issues/582#issuecomment-1883074908
+         */
+        const firstChild = rowRef.current.firstElementChild;
+        const height = firstChild
+          ? firstChild.clientHeight
+          : rowRef.current.clientHeight;
+
+        setItemSize(height);
+      }
+    }, [index, rowRef, setItemSize]);
+
     const baseOption = data[index];
+    const { key, props } = baseOption;
+
     /**
      * react-window calculates the absolute positions of the list items, via an inline style, so
      * we need to add it to each list item that is being rendered in the viewable list window.
-     * See here if you need to know more: https://github.com/bvaughn/react-window?tab=readme-ov-file#why-is-my-list-blank-when-i-scroll
+     * @see here if you need to know more: https://github.com/bvaughn/react-window?tab=readme-ov-file#why-is-my-list-blank-when-i-scroll
      */
-    const optionItem = { ...baseOption, props: { ...baseOption.props, style } };
-    return optionItem;
+    return (
+      <div ref={rowRef}>
+        <li
+          style={{
+            ...style,
+            height: "auto",
+          }}
+          key={key}
+          {...props}
+        />
+      </div>
+    );
```

Comment:
We should memoize anything using `style`. Also, we should probably put `{...props}` first. It's rare that you would define props on a component and then overwrite them:

```suggestion
    return (
      <div ref={rowRef}>
        <li
          {...props}
          key={key}
          style={styles}
        />
      </div>
    );
```